Data/models/FaviconMO.swift

```diff
@@ -19,24 +19,14 @@ public class FaviconMO: NSManagedObject {
     }
 
     public class func get(forFaviconUrl urlString: String, context: NSManagedObjectContext) -> FaviconMO? {
-        let fetchRequest = NSFetchRequest<NSFetchRequestResult>()
-        fetchRequest.entity = FaviconMO.entity(context)
-        fetchRequest.predicate = NSPredicate(format: "url == %@", urlString)
-        var result: FaviconMO? = nil
-        do {
-            let results = try context.fetch(fetchRequest) as? [FaviconMO]
-            if let item = results?.first {
-                result = item
-            }
-        } catch {
-            let fetchError = error as NSError
-            print(fetchError)
-        }
-        return result
+        let urlKeyPath = #keyPath(FaviconMO.url)
+        let predicate = NSPredicate(format: "\(urlKeyPath) == %@", urlString)
```

Comment:
May as well stick to some convention here. Probably:
`NSPredicate(format: "\(urlKeyPath) == \(urlString)")`

Reply:
I think your solution poses risk of sql injection attack. You could append stuff like `' OR 1 == 1` to bypass some checks. Queries with `%@` arguments are safer. Ofc this is only a Favicon table so not a big deal, but we should stick to those parametrized predicates.

Same applies to your other NSPredicate comments.